_posts/2020-06-07-higher-order-hydras.markdown

```diff
@@ -0,0 +1,491 @@
+---
+layout: post
+title: "Amiko hydras"
+date: 2020-06-07 23:00
+categories: math ordinals
+---
+
+<div class="no-js">
+Math on this page may not display correctly without JavaScript.
+</div>
+
+{::options parse_block_html="true" /}
+
+# Kirby-Paris and Buchholz hydras
+{: #kp-buchholz-hydra}
+
+{% include collapser.markdown %}
+
+<div>
+
+Hydras are a classic iterative game played on trees, known by googologists for the fast-growing functions derived from them and by serious mathematicians for their contributions to ordinal analysis.
+Unlike their combinatorial counterparts like TREE and SCG, no search is required to compute these fast-growing function values - just keep applying the transformation rule to the tree until the game says to stop.
+
+Kirby-Paris hydras came first.
+For Kirby-Paris hydras, the rules are simple.
+Start with your hydra, which is an unordered unlabelled rooted tree.
+At each stage, choose a leaf node \\\(C\\\) to chop and a non-negative integer \\\(N\\\).
+If \\\(C\\\) is a child of the root, it is removed from the tree and that's it.
+Otherwise, let \\\(P\\\) be \\\(C\\\)'s parent, and \\\(G\\\) be \\\(P\\\)'s parent.
+Remove \\\(C\\\) from \\\(P\\\), then add \\\(N\\\) copies of the modified \\\(P\\\) as children to \\\(G\\\).
+The game ends when the hydra is reduced to a single node.
+
+To obtain a fast-growing function, we can fix \\\(N\\\), say, \\\(N = 1\\\) at the first step, then \\\(N = 2\\\), \\\(N = 3\\\), and so on,
+and decide on a simple rule for where to cut, say, always choosing the rightmost leaf.
```

Comment:
In the sentence fragment "and decide on a simple rule for where to cut, say, always choosing the rightmost leaf.", I would switch "say" for "for example", to bring more variation to reading.

Reply:
I think that would add more confusion than is worth it, since the only other option really is to always choose the leftmost leaf, which is not functionally different from always choosing the rightmost leaf (just flip the hydra).

Reply:
That's a good point, I didn't even consider that!